AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -2255,7 +2254,11 @@ function New-ADAccountForStorageAccount {
     Write-Verbose -Message "ObjectType: $ObjectType"
 
     if ([System.String]::IsNullOrEmpty($Domain)) {
-        $domainInfo = Get-ADDomain
+        if ($ObjectType -ieq "ComputerAccount") {
+            $domainInfo = Get-ADDomain -Current LocalComputer
+        } else { # "ServiceLogonAccount"
+            $domainInfo = Get-ADDomain -Current LoggedOnUser
+        }

```

Comment:
I'm not sure if we should change behavior now that if no domain and no OU is specified, we create the account in the domain of the user rather than the domain of the computer.  i'm sure it's a rare scenario but could be confusing.    @wmgries - would be good to have your opinion - what do you think about this one?  We could also give a message in the case that the user and the computer are in a different domain and then tell the user to rerun with "domain" specified.

Reply:
Yeah, I'm not sure if I would do this. I don't think it's hugely harmful, just is harder to understand the behavior.

Reply:
The reasons are:
1.	This is to align to the requirement where the default OU should be chosen based on the object type. When the object type is “ComputerAccount”, we should lookup the computer object to find the OU, and to look up the computer object we should use the domain of the computer instead of the domain of the user.
2.	There is a customer feedback complaining that the storage account was put into the directory of his user account by default, which was not his intention. So I guess just using the user’s domain as the default may not be what the customer wants.